substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/StructureTypeEntry.java

```diff
@@ -45,9 +48,25 @@ public abstract class StructureTypeEntry extends TypeEntry {
      */
     protected final List<FieldEntry> fields;
 
+    /**
+     * The type signature of this types' layout.
+     */
+    protected long layoutTypeSignature;
+    protected long indirectLayoutTypeSignature;
```

Comment:
Do we really need a compressed type variant `layoutTypeSignature`?

Reply:
No, the only time we use it currently is in the type unit for the compressed type, so we can move it there.